packages/compass-aggregations/src/components/stage-editor/stage-editor.jsx

```diff
@@ -65,7 +65,19 @@ export class StageEditor extends PureComponent {
       // Focus the editor when the stage operator has changed.
       this.editor.focus();
     }
-    // TODO: try showing annotation with syntax error?
+    if (this.props.syntaxError && this.props.syntaxError.loc) {
+      const { line: row, column } = this.props.syntaxError.loc;
+      this.editor?.getSession().setAnnotations([
+        {
+          row: row - 1,
+          column,
+          text: this.props.syntaxError.message,
+          type: 'error'
+        }
+      ]);
```

Comment:
Adding them here to test now this annotations api is working. This was not in the scope, but I figured we can keep it? Can also remove if you feel like we should not expand the scope here

<img width="398" alt="image" src="https://user-images.githubusercontent.com/5036933/196686406-f8e95f5a-faa0-4d17-b2fe-6a5416903350.png">

Reply:
this looks really good. not sure if we need it in stage editor, but in textual it will look good.